### PR TITLE
Fixes weird turret warning messages issue for blind/deaf characters

### LINKED ISSUE #46058: Player shouldn't be able to hear a automated turret emit an IFF warning beep if they're deafened.
I'm wearing earplugs (and therefore have the `Deaf!` status effect) to protect my hearing from the incredible noise of my Raid SUV's mounted .50 cal shooting down errant zombies. Yet somehow I have many instances of `The ++ mounted M2 Browning emits an IFF warning beep.` in my log.

# Steps To Reproduce

1. Have a turret mounted weapon with a turret control unit installed and enabled. Powered, loaded with ammo, and whatnot.
2. Wear ear protection that deafens you, or somehow become deafened.
3. Contrive to stand between your mounted turret and a valid target for the turret.
4. See the "warning beep" printed in the event log.

# Expected behavior

No printed warning message in the event log whatsoever. The player shouldn't be able to benefit from things that are sound only if they're also benefiting from being deaf. 

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-8201-gc49b2cc [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Mythical Martial Arts [MMA],
    Alternative Map Key [alt_map_key],
    Magiclysm [magiclysm],
    Graphical Overmap [Graphical_Overmap],
    Graphical Overmap Magiclysm [Graphical_Overmap_Magiclysm],
    No Fungal Monsters [No_Fungi]
]

### PR DESCRIPTION
**Pull request by SkyBlueFlash**
#### Summary

SUMMARY: Bugfixes "Added a check to see if the player is deaf or blind in order to print the correct warning message when the player is standing between mounted turret and its target"

#### Purpose of change

This commit fixes #46058

Before the commit the player was able to hear a warning beep even when deaf and see the turret pointing toward them even when blind.

Steps to reproduce:

    Have a turret mounted weapon with a turret control unit installed and enabled. Powered, loaded with ammo, and whatnot.
    Wear ear protection that deafens you, or somehow become deafened.
    Contrive to stand between your mounted turret and a valid target for the turret.
    See the "warning beep" printed in the event log.

#### Describe the solution

Added the variable "u_hear" similar to the already implemented variable "u_see" then added multiple if-else statements to check if the player can:

see and hear: print warning beep and tell the player that the turret pointed in their direction.
see but cant hear: only tell the player that the turret pointed in their direction.
hear but cant see: only print warning beep.
neither see or hear: print nothing

#### Testing

This fix was tested in-game by spawning a vehicle, equipping it with a turret and then standing between it and its target while blind and/or deaf, and it appear to solve the issue.

### CODE REVIEW

File: src/turret.cpp [hunk 1]
```diff
@@ -589,16 +590,20 @@ int vehicle::automatic_fire_turret( vehicle_part &pt )
...
                              cpu.name, boo_hoo );
-                } else {
+                } else if( !u_see & u_hear ) {
                     add_msg( m_warning, ngettext( "%s emits an IFF warning beep.",
                                                   "%s emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
+                } else if( u_see & !u_hear ) {
+                    add_msg( m_warning, "%s points in your direction",
+                             cpu.name, boo_hoo );
```
Comment (BrettDong) [inline src/turret.cpp:606]: 1. Missing period: `"%s points in your direction."`
2. There is only one `%s` but you provided `cpu.name` and `boo_hoo`. I guess only `cpu.name` is used here, is it right?

```suggestion
                    add_msg( m_warning, "%s points in your direction.", cpu.name );
```

Comment (Qrox) [reply src/turret.cpp:606]:
```suggestion
                    add_msg( m_warning, _( "%s points in your direction" ), cpu.name );
```
The string should also be translated.

File: src/turret.cpp [hunk 2]
```diff
@@ -589,16 +590,20 @@ int vehicle::automatic_fire_turret( vehicle_part &pt )
...
                     add_msg( m_warning, ngettext( "%s points in your direction and emits an IFF warning beep.",
                                                   "%s points in your direction and emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
-                } else {
+                } else if( !u_see & u_hear ) {
                     add_msg( m_warning, ngettext( "%s emits an IFF warning beep.",
                                                   "%s emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
```
Comment (Night-Pryanik) [inline src/turret.cpp:603]: I think we should rewrite the text for this section. In this case we can't see what's beeping, right? So the text should be something like `You hear a beep.` and `You hear %d annoyed sounding beeps.`

Comment (SkyBlueFlash) [reply src/turret.cpp]: I made some changes based on your feedback please take a look when you have some time.
sorry for taking too much time and making too many commits on something that is supposed to be simple, this is my first contribution in a large project and it appears I got a little too confident.

Comment (Night-Pryanik) [reply src/turret.cpp]: Looks good to me. Good work.
Even a simple change could take a lot of time, especially for first-time contributor, so no need to apologize.

**Review by BrettDong — APPROVED** [review]
Tested locally and all three branches function normally.

**Pull request merged**